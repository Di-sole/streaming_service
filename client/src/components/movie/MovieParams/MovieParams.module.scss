.container {
    width: 100%;

    &__list {
        display: flex;
        align-items: center;
        justify-content: center;
        color: #ffffffcc;
        margin: 0 0 4px 0;
        flex-wrap: wrap;

        &:last-child {
            margin: 0;
        }
    }

    &__listItem {
        position: relative;
        font-size: 15px;
        line-height: 20px;
        vertical-align: top;
        margin: 0px 8px 0 8px;

        &:last-child {
            margin: 0;
        }

        &:first-child::before {
            content: none;
        }

        &_point {
            display: inline-flex;
            align-items: center;
            margin: 0;

            &::before {
                content: '';
                width: 2px;
                height: 2px;
                background-color: #ffffffb8;
                margin: 5px;
            }

            &:first-child::before {
                content: none;
            }
        }
    }

    &__listIcon {
        width: 15px;
        height: 15px;
        margin: 0 5px 0 0;

        &_sound {
            margin: 0 -3px 0 5px;
        }
    }
}

@media (max-width: 1159px) {
    .container {
        &__list {
            justify-content: start;
        }
    }
}

@media (max-width: 879px) {
    .container {
        &__list {
            justify-content: center;
        }
    }
}
