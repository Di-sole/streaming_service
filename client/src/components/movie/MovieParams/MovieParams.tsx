import styles from './MovieParams.module.scss';
import TextLinkUI from '../../UI/links/TextLink/TextLinkUI';
import TextBadge from '../../UI/badges/TextBadge/TextBadge';
import Image from 'next/image';
import soundIcon from '../../../../public/icons/sound.png';
import subtitlesIcon from '../../../../public/icons/subtitles.png';
import IMovie from '@/models/IMovie';
import { firstCapitalLetter, minutesToHours } from '../../../utils/functions';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';

interface MovieParamsProps {
    movie: IMovie;
}

const MovieParams = ({ movie }: MovieParamsProps) => {
    const { locale } = useRouter();
    const { t } = useTranslation('movie');
    return (
        <div className={styles.container} data-testid="div-params">
            <div className={styles.container__list}>
                <TextLinkUI href={`/collections/${movie.year}`} option="bright">
                    {movie.year}
                </TextLinkUI>
                <span className={styles.container__listItem}>
                    {minutesToHours(movie.movieLength, [t('textTime.0'), t('textTime.1')])}
                </span>
                <span className={styles.container__listItem}>{movie.ageRating}+</span>
            </div>
            <div className={styles.container__list}>
                <TextLinkUI
                    href={`/collections/${movie.countries[0].enName.split(' ').join('_')}`}
                    option="bright"
                    className={[styles.container__listItem, styles.container__listItem_point].join(' ')}
                >
                    {locale === 'ru' ? movie.countries[0].name : movie.countries[0].enName}
                </TextLinkUI>
                {movie.genres.slice(0, 3).map((genre) => (
                    <TextLinkUI
                        key={genre.id}
                        href={`/collections/${genre.enName}`}
                        option="bright"
                        className={[styles.container__listItem, styles.container__listItem_point].join(' ')}
                    >
                        {firstCapitalLetter(locale === 'ru' ? genre.name : genre.enName)}
                    </TextLinkUI>
                ))}
            </div>
            <div className={styles.container__list}>
                <TextBadge text="FullHD" />
                <Image
                    src={soundIcon}
                    alt="soundIcon"
                    className={[styles.container__listIcon, styles.container__listIcon_sound].join(' ')}
                />
                <div className={styles.container__listItem}>
                    {['Рус', 'Eng'].map((voice) => (
                        <div
                            key={voice}
                            className={[styles.container__listItem, styles.container__listItem_point].join(' ')}
                        >
                            {voice}
                        </div>
                    ))}
                </div>

                <Image src={subtitlesIcon} alt="subtitlesIcon" className={styles.container__listIcon} />
                <div className={styles.container__listItem}>
                    <div className={[styles.container__listItem, styles.container__listItem_point].join(' ')}>Рус</div>
                </div>
            </div>
        </div>
    );
};

export default MovieParams;
