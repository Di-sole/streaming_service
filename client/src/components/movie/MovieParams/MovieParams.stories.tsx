import { Meta, StoryObj } from '@storybook/react';
import MovieParams from './MovieParams';
import './MovieParams.module.scss';
import IMovie from '@/models/IMovie';

const meta: Meta<typeof MovieParams> = {
    title: 'Information/Params',
    component: MovieParams,
};

export default meta;

type Story = StoryObj<typeof MovieParams>;

const data: IMovie = {
    id: 1,
    type: 'movie',
    name: 'Название',
    enName: 'Название',
    posterUrl: '/img/posterTest.jpg',
    posterPreviewURL: '/img/posterTest.jpg',
    year: 1900,
    description: '',
    shortDescription: '',
    ageRating: 0,
    ratingKp: 9.5,
    votesKp: 11111,
    movieLength: 126,
    genres: [
        { id: 1, name: 'Жанр', enName: 'Жанр' },
        { id: 2, name: 'Жанр', enName: 'Жанр' },
    ],
    countries: [{ id: 1, name: 'Страна', enName: 'Страна' }],
    persons: [],
    trailer: '',
    watchingWithMovie: [],
    comments: [],
};

export const Params: Story = {
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        movie: data,
    },
};
