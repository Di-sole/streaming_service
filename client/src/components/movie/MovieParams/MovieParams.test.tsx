import { render, screen } from '@testing-library/react';
import MovieParams from './MovieParams';
import { movieTest } from '../../../testAsserts/testItems';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('MOVIEPARAMS SNAPSHOTS TESTS', () => {
    test('MovieParams should not be changed', () => {
        render(<MovieParams movie={movieTest} />);

        const div = screen.getByTestId('params');
        expect(div).toMatchSnapshot();
    });
});
