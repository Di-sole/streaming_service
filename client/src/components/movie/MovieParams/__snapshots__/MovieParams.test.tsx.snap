// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIEPARAMS SNAPSHOTS TESTS MovieParams should not be changed 1`] = `
<div
  class="container"
  data-testid="params"
>
  <div
    class="container__list"
  >
    <a
      class="container container_bright"
      href="/collections/2011"
    >
      2011
    </a>
    <span
      class="container__listItem"
    >
      1 textTime.0 52 textTime.1
    </span>
    <span
      class="container__listItem"
    >
      16
      +
    </span>
  </div>
  <div
    class="container__list"
  >
    <a
      class="container container_bright container__listItem container__listItem_point"
      href="/collections/France"
    >
      France
    </a>
    <a
      class="container container_bright container__listItem container__listItem_point"
      href="/collections/drama"
    >
      Drama
    </a>
    <a
      class="container container_bright container__listItem container__listItem_point"
      href="/collections/comedy"
    >
      Comedy
    </a>
    <a
      class="container container_bright container__listItem container__listItem_point"
      href="/collections/biography"
    >
      Biography
    </a>
  </div>
  <div
    class="container__list"
  >
    <div
      class="badge"
      data-testid="textBadge"
    >
      FullHD
    </div>
    <img
      alt="soundIcon"
      class="container__listIcon container__listIcon_sound"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <div
      class="container__listItem"
    >
      <div
        class="container__listItem container__listItem_point"
      >
        Рус
      </div>
      <div
        class="container__listItem container__listItem_point"
      >
        Eng
      </div>
    </div>
    <img
      alt="subtitlesIcon"
      class="container__listIcon"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <div
      class="container__listItem"
    >
      <div
        class="container__listItem container__listItem_point"
      >
        Рус
      </div>
    </div>
  </div>
</div>
`;
